README.md

```diff
@@ -291,6 +291,57 @@ writeResultFile("hostFunction.result", hostFunctionResult);
 ```
 -->
 
+### Store and Instantiating Multiple Modules
+
+A [Store][spec] is an intermediate-level abstraction that collects Wasm function, global, memory, and table instances
+as named entities.
+
+It simplifies creating instances when there are a lot of interdependencies, by collecting all the
+
+In the simplest case, it allows to register single host functions, globals, memories and tables:
+
+```java
+import com.dylibso.chicory.runtime.Store;
+
+// instantiate the store
+var store = new Store();
+// registers `console.log` defined above in the store
+store.addHostFunction(func);
+```
+
+However, the store also automatically exposes the exports of a module to the other instances that are registered.
+
+```java
+// registers `logger` into the store
+store.register("logger", instance);
```

Comment:
where `instance` is coming from?

Reply:
from above 😬

Reply:
do you want me to show the instantiation in the same code snippet, or is it ok to clarify it's the `instance` defined above? notice the same issue is with `console.log` (`func`) a few lines earlier.

Reply:
good question, no strong opinion, either of your proposals work for me.

Reply:
clarified that `func` and `instance` are defined in the previous section of the README

Reply:
thanks :+1: